### ISSUE TITLE
mm_x and mm_y not adjusted when resampling image

### ISSUE THREAD

**Issue by YoniSchirris** [girder/large_image#439]
## Issue summary
`tileIterator`'s dictionary contains `mm_x` and `mm_y` which is defined to be *"mm_x, mm_y: size of the current tile pixel in millimeters."* in `base.py`. When a different-than-maximum magnification is requested, the `mm_x` and `mm_y` do not change accordingly, and it thus denotes the "size of pixel in millimeters for maximum magnification" instead of that of "the current tile"

## Current output
When creating a tileIterator for a whole-slide image with 0.2505 microns per pixel (through openslide: `slide.properties[openslide.PROPERTY_NAME_MPP_X]`)
```python
slide = openslide.OpenSlide("path/to/WSI")
print(slide.properties[openslide.PROPERTY_NAME_MPP_X])  # 0.2505

li_slide = large_image.getTileSource("path/to/WSI")
tile_iterator = li_slide.tileIterator(tile_size=dict(width=225, height=225),
                                                  scale=dict(mm_x=0.5e-3, mm_y=0.5e-3),
                                                  region=dict(left=0, top=0, right=region_width, bottom=region_height),
                                                  tile_overlap=dict(x=0, y=0),
                                                  format=large_image.tilesource.TILE_FORMAT_NUMPY,
                                                  resample=PIL.Image.BICUBIC)
```

we get tiles with a dictionary like:

```json
{
        "x": 11674.0,
        "y": 898.0,
        "width": 224,
        "height": 224,
        "level": 9,
        "level_x": 26,
        "level_y": 2,
        "magnification": 20.040242969658358,
        "mm_x": 0.0002505,
        "mm_y": 0.0002505,
        "tile_position": {
            "level_x": 26,
            "level_y": 2,
            "region_x": 26,
            "region_y": 2,
            "position": 420
        },
        "iterator_range": {
            "level_x_min": 0,
            "level_y_min": 0,
            "level_x_max": 197,
            "level_y_max": 64,
            "region_x_max": 197,
            "region_y_max": 64,
            "position": 12608
        },
        "tile_overlap": {
            "left": 0,
            "top": 0,
            "right": 0,
            "bottom": 0
        },
        "format": "numpy",
        "gx": 11674.0,
        "gy": 898.0,
        "gwidth": 449.0,
        "gheight": 449.0,
        "scaled": 0.5010060742414589,
        "tile_x": 11674,
        "tile_y": 898,
        "tile_width": 449,
        "tile_height": 449,
        "tile_magnification": 40.0
    }
```

where the important entries to note are:

```json
        "width": 224,
        "height": 224,
        "magnification": 20.040242969658358,
        "mm_x": 0.0002505,
        "mm_y": 0.0002505,
        "tile_width": 449,
        "tile_height": 449,
        "scaled": 0.5010060742414589,
        "tile_magnification": 40.0
```

Meaning the tile is correctly sampled and is downscaled from magnification 40 (with mm_x=0.2505)  to 20.04 (mm_x=0.0002505/0.50100..=0.0005), yet this is not clearly shown in the dictionary. (Which took me 6+ hours of sanity checks to ensure it was going correctly)

## Expected result:
Similarly to magnification, width, height, x, and y, I would expect two entries: one from before and after resampling, e.g. (taking only the small 'important' json from above):
```json
        "width": 224,
        "height": 224,
        "magnification": 20.040242969658358,
        "mm_x": 0.0005,
        "mm_y": 0.0005,
        "tile_width": 449,
        "tile_height": 449,
        "tile_mm_x": 0.0002505,
        "tile_mm_y": 0.0002505,
        "scaled": 0.5010060742414589,
        "tile_magnification": 40.0
```

**Comment by manthey** [comment]
Thanks for submitting this.  I'll try to get a PR this week.

**Comment by YoniSchirris** [comment]
One additional note:
- It does not only display maximum resolution, it does display the `mm_x` and `mm_y` for an exact magnification level correctly
- However, when resampled, this is not adjusted

**Comment by manthey** [comment]
See PR #441.  You'll notice inexact values, so when you ask for 0.0006 mm you'll get results that are slightly different, e.g., 0.00059987.   This should be addressed when #440 is resolved.

**Comment by YoniSchirris** [comment]
Slight variations are acceptable I'd say as indeed we are working with integer tile sizes, so an exact match might not be possible. Thanks for the fix, and the addition of the test in your PR looks nice.
